Backend/api/user/router.js

```diff
@@ -5,31 +5,48 @@ const {
   checkAdminAuth,
 } = require("../middleware/auth-validation");
 const {
+  validationResults,
   isUserValid,
-  valdiationResults,
+  isWorkoutValid,
 } = require("../middleware/body-validation");
 
 const router = require("express").Router();
 
-router.get("/:id?", checkUserAuth, userController.getUser);
+router.get("/get/:_id?", checkUserAuth, userController.getUser);
 
 router.post(
   "/register",
   isUserValid,
-  valdiationResults,
+  validationResults,
   userController.register
 );
 
 router.post("/login", userController.login);
 
 router.put(
-  "/:_id",
+  "/update/:_id?",
   checkUserAuth,
   isUserValid,
-  valdiationResults,
+  validationResults,
   userController.updateUser
 );
 
 router.delete("/:_id", checkAdminAuth, userController.deleteUser);
 
+router.get("/getWorkouts", checkUserAuth, userController.getWorkouts);
+
+router.put(
+  "/createWorkout",
```

Comment:
This is not intuitive for the client. Ideally, operations for a resource type should be handled by routes with similar name.
Eg:
GET `/:userId/workouts` - Gets workouts for a user
GET `/:userId/workouts/:workoutId` - Gets a specific workout for a user
POST `/:userId/workouts` - Creates a workout for a user
PUT `/:userId/workouts/:workoutId` - Updates a user's workout
DELETE `/:userId/workouts/:workoutId` - Deletes a user's workout

This is a more intuitive way to use the API for the clients, while preserving the idempotence of HTTP methods.
You can read more about HTTP methods here: https://restfulapi.net/idempotent-rest-apis/